code.rb

```diff
@@ -0,0 +1,178 @@
+module Enumerable
+  def my_each
+    size = self.size
+    index = 0
+
+    while index < size
+
+      yield(self[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    size = self.size
+    index = 0
+
+    while index < size
+
+      yield(self[index], index)
+      index += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+
+    my_each do |x|
+      new_array.push(x) if yield(x)
+    end
+
+    new_array
+  end
+
+  def my_all?(val = nil)
+    all_true = true
+
+    if block_given?
+
+      my_each do |x|
+        all_true = false unless yield(x)
+      end
+
+    elsif val
+
+      my_each do |x|
+        all_true = false unless x == val
+      end
+
+    else
+
+      my_each do |x|
+        all_true = false unless x
+      end
+
+    end
+
+    all_true
+  end
+
+  def my_any?(val = nil)
+    any = false
+
+    if block_given?
+      my_each do |x|
+        any = true if yield(x)
+      end
+
+    elsif val
+      my_each do |x|
+        any = true if x == val
+      end
+
+    else
+
+      my_each do |x|
+        any = true if x
+      end
+
+    end
+
+    any
+  end
+
+  def my_none?(val = nil)
+    none = true
+
+    if block_given?
+      my_each do |x|
+        none = false if yield(x)
+      end
+
+    elsif val
+
+      my_each do |x|
+        none = false if x == val
+      end
+
+    else
+
+      my_each do |x|
+        none = false if x
+      end
+
+    end
+
+    none
+  end
+
+  def my_count(val = nil)
+    total = 0
+
+    if block_given?
+      my_each do |x|
+        total += 1 if yield(x)
+      end
+
+    elsif val
+
+      my_each do |x|
+        total += 1 if x == val
+      end
+
+    else
+
+      total = length
+
+    end
+
+    total
+  end
+
+  def my_map(my_proc = nil)
+    new_array = []
+
+    if my_proc
+      my_each do |x|
+        new_array.push(my_proc.call(x))
+      end
+
+    else
+
+      my_each do |x|
+        new_array.push(yield(x))
+      end
+    end
+
+    new_array
+  end
+
+  def my_inject(init = nil, proc = nil)
```

Comment:
Your `my_inject` does not quite work as should.
Please see a screenshot of failed tests below:
![Screen Shot 2020-01-16 at 4 41 49 PM](https://user-images.githubusercontent.com/31193823/72561489-b82a3f80-387f-11ea-934e-b14cf70559ff.png)

You can read more about the original `inject` method in this [documentation](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-inject)